### PR TITLE
How to write an Ingress controller docs

### PR DESCRIPTION
**Pull request by bprashanth**
Does what each of the commits say, main goal is to describe how one writes an Ingress controller with a simple example.

### CODE REVIEW

**Comment by bprashanth** [comment]
@brendandburns can you reivew if you're not too busy with the release? since Tim is out for a few weeks. There's already a link leading here from user docs, but it's currently pretty bare.

**Comment by bprashanth** [comment]
Pretty sure Brendan's busy with the release. Mike danese to the rescue.

File: Ingress/controllers/nginx-alpha/rc.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  replicas: 1
+  selector:
+    app: nginx-ingress
+  template:
+    metadata:
+      labels:
+        app: nginx-ingress
+    spec:
+      containers:
+      - image: gcr.io/google_containers/nginx-ingress:0.1
```
Comment (bprashanth) [inline Ingress/controllers/nginx-alpha/rc.yaml]: need to push this out before checkin

File: Ingress/controllers/README.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		log.Fatalf("Failed to open %v: %v", nginxConf, err)
+	} else if err := tmpl.Execute(w, ingresses); err != nil {
+		log.Fatalf("Failed to write template %v", err)
+	}
+	shellOut("nginx -s reload")
+}
+```
+
+All this is doing is:
+* List Ingresses, optionally you can watch for changes (see [GCE Ingress controller](https://github.com/kubernetes/contrib/blob/master/Ingress/controllers/gce/controller.go)i for an example)
```
Comment (mikedanese) [inline Ingress/controllers/README.md]: extra "i" at the end of the GCE Ingress controller link

File: Ingress/controllers/README.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+Configuring a webserver or loadbalancer is harder than it should be. Most webserver configuration files are very similar. There are some applications that have weird little quirks that tend to throw a wrench in things, but for the most part you can apply the same logic to them and achieve a desired result. The Ingress resource embodies this idea, and an Ingress controller is meant to handle all the quirks associated with a specific "class" of Ingress (be it a single instance of a loadbalancer, or a more complicated setup of frontends that provide GSLB, DDoS protection etc).
+
+## What is an Ingress Controller?
+
+An Ingress Controller is a daemon, deployed as a Kubernetes Pod, that watches the ApiServer's `/ingresses` endpoint for updates to the [Ingress resource](https://github.com/kubernetes/kubernetes/blob/master/docs/user-guide/ingress.md). Its job is to satisfy requests for ingress.
+
+## Writing an Ingress Controller
+
+Writing an Ingress controller is simple. By way of example, the [nginx controller] (nginx-alpha) does the following:
+* Poll till apiserver reports a new Ingress
```
Comment (mikedanese) [inline Ingress/controllers/README.md]: s/till/until/ is formal

File: Ingress/controllers/README.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    resolver 127.0.0.1;
+
+    location /bar {
+      proxy_pass http://barSvc;
+    }
+  }
+}
+```
+
+And you can reach the `/foo` and `/bar` endpoints on the publicIP of VM the nginx-ingress pod landed on.
```
Comment (mikedanese) [inline Ingress/controllers/README.md]: "of the VM", missing a the

File: Ingress/controllers/nginx-alpha/README.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    targetPort: 8080
+    protocol: TCP
+    name: http
+  selector:
+    app: echoheaders
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: echoheadersy
```
Comment (mikedanese) [inline Ingress/controllers/nginx-alpha/README.md]: "echeadersy"? is the "y" intentional or a typo?

**Comment by mikedanese** [comment]
Very minor comments. LGTM otherwise.

**Comment by bprashanth** [comment]
Done, PTAL

**Comment by mikedanese** [comment]
LGTM feel free to merge when image is pushed

**Comment by bprashanth** [comment]
Thanks!

**Pull request merged**